#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "voxel_classifier"

gen = ParameterGenerator()

gen.add(
    "cluster_minimum_points",
    int_t,
    0,
    "The minimum number of neighbours a point needs to be considered as a core point.",
    3,
    1,
    50
)

gen.add(
    "cluster_epsilon",
    double_t,
    0,
    "Maximum distance between two points for them to be considered neighbours.",
    0.50,
    0.0001,
    10
)

gen.add(
    "color_weight",
    double_t,
    0,
    "Weight of the color of voxels, from 0% to 200%",
    0,
    0,
    2
)

exit(gen.generate(PACKAGE, "voxel_classifier", "voxel_classifier"))
